src/test_l3_dsHdmiIn.c

```diff
@@ -0,0 +1,1461 @@
+*/
+void test_l3_HdmiIn_set_edid2allmsupport(void)
+{
+    gTestID = 16;
+    UT_LOG("\n In %s [%02d%03d]\n", __FUNCTION__, gTestGroup, gTestID);
+
+    dsError_t ret   = dsERR_NONE;
+    dsHdmiInPort_t port = dsHDMI_IN_PORT_MAX;
+    int32_t select;
+    bool allmsupport = false;
+    uint8_t numInputPorts = 0;
+
+    numInputPorts = UT_KVP_PROFILE_GET_UINT8("dsHdmiIn/numberOfPorts");
+
+    printf(" \n================Please Select Inputs==================\n");
+
+    printf("\n*******Enter the port number to select*******\n");
+    scanf("%d", &select);
+    readAndDiscardRestOfLine(stdin);
+
+    if(select <= 0 || select > numInputPorts)
+    {
+       printf("\nInvalid port selected\n");
+       goto exit;
+    }
+
+    port = select;
+
+    printf("\n*******Enter the allmsupport to select*******\t"
+                     "Acceptable inputs are:\t"
+                     "true (i.e 1)\t"
+                     "false(i.e 0)\t"
+             "******************************************\n");
+    scanf("%d", &select);
+    readAndDiscardRestOfLine(stdin);
+    if(select < 0 || select > 1)
+    {
+       printf("\nInvalid allmsupport value selected\n");
+       goto exit;
+    }
+  
+    allmsupport = (bool)select;
+
+    ret = dsSetEdid2AllmSupport(port, allmsupport);
+    if(ret != dsERR_NONE)
+    {
+       UT_LOG_ERROR("Result: Failed dsSetEdid2AllmSupport Error:[%s]",ut_control_GetMapString(dsError_mapTable, ret));
+    } else {
+           UT_LOG_INFO("Result : dsSetEdid2AllmSupport port type: %s, portnumber: %s\t"
+                          "allmsupport set: %s",
+                 ut_control_GetMapString(dsHdmiInPort_mapTable, port), port,
+                 ut_control_GetMapString(bool_mapTable, allmsupport));
+    }
+
+exit:
+UT_LOG_INFO("Out %s", __FUNCTION__);
+}
+
+/**
+* @brief This test to get allmsupport.
+*
+* This test function gets allmsupport of HdmiInput port on platform.
+*
+* **Test Group ID:** 03@n
+* **Test Case ID:** 017@n
+*
+* **Test Procedure:**
+* Refer to Test specification documentation
+* [ds-hdmi-in_halSpec.md](../../docs/pages/ds-hdmi-in_halSpec.md)
+*/
+void test_l3_HdmiIn_get_edid2allmsupport(void)
+{
+    gTestID = 17;
+    UT_LOG("\n In %s [%02d%03d]\n", __FUNCTION__, gTestGroup, gTestID);
+
+    dsError_t ret   = dsERR_NONE;
+    dsHdmiInPort_t port = dsHDMI_IN_PORT_MAX;
+    int32_t select;
+    bool allmsupport = false;
+    uint8_t numInputPorts = 0;
+
+    numInputPorts = UT_KVP_PROFILE_GET_UINT8("dsHdmiIn/numberOfPorts");
+
+    printf(" \n================Please Select Inputs==================\n");
+
+    printf("\n*******Enter the port number to select*******\n");
+    scanf("%d", &select);
+    readAndDiscardRestOfLine(stdin);
+
+    if(select <= 0 || select > numInputPorts)
+    {
+       printf("\nInvalid port selected\n");
+       goto exit;
+    }
+
+    port = select;
+
+    ret = dsGetEdid2AllmSupport(port, &allmsupport);
+    if(ret != dsERR_NONE)
```

Comment:
Remove the else case

In all cases, you need to see the results output

You've got some serious debugging issues todo if this was to assert and a failure of L1-L2 to fix in that case.

Reply:
done